tutorials/truffle-dev-life-cycle.md

```diff
@@ -0,0 +1,538 @@
+
+## Write Tests {: #write-tests }
+
+Before sending our code out into the wild, we'll want to test our smart contracts to ensure they function as expected. Truffle provides the option of writing tests in JavaScript, TypeScript, or Solidity. It also comes with out-of-the-box support for [Mocha](https://mochajs.org/){target=_blank} and [Chai](https://chaijs.com/){target=_blank}.
+
+For this guide, we'll write our tests in JavaScript so we can take advantage of the built-in support for Mocha and Chai. 
+
+If you're familiar with Mocha, you're probably used to using the `describe` function to group tests and the `it` function for each individual test. When writing tests with Truffle, you'll replace `describe` with `contract`. The `contract` function is exactly like `describe`, but it includes additional functionality that will re-deploy your migrations at the beginning of every test file, providing a clean-room environment. You'll still use the `it` function like you normally would for the individual tests.
+
+Truffle also makes testing easier by including a `web3` instance in each test file that is configured for the correct network, so you don't have to configure anything yourself. You'll simply run `npx truffle test --network <network-name>`.
+
+### Test Setup {: #test-setup }
+
+Before getting started, we'll need to generate artifacts for our contract, which will be used in our tests. We can run the following command:
+
+```
+npx truffle compile
+```
+
+The artifacts will be written to the `build/contracts` directory. Next, we can add our test file, which will start with `test_` to indicate that it's a test file:
+
+```
+touch test test_NftMarketplace.js
+```
+
+Now that we can set up our test file, let's take a minute to review what we'll need to do next:
+
+- Import the artifacts for the `NftMarketplace` and `DizzyDragons` contracts using Truffle's `artifacts.require()`, which provides an abstraction instance of a contract
+- Create a `contract` function to group our tests. The `contract` function will also provide us with our account we have setup in our `truffle-config.js` file. As we used the Moonbeam Truffle box, our development account has been set up for us. When we move on to deploy and test our contracts on Moonbase Alpha and Moonbeam, we'll need to configure our accounts
+- For each test, we're going to need to deploy our contracts and mint an NFT. To do this, we can take advantage of the [`beforeEach` hook provided by Mocha](https://mochajs.org/#hooks){target=_blank}
+- As we'll be minting an NFT for each test, we'll need to have a `tokenUri`. The `tokenUri` that we'll use for our examples will be for Daizy, our first Dizzy Dragon NFT. The `tokenUri` will be set to `"https://gateway.pinata.cloud/ipfs/QmTCib5LvSrb7sshLhLvzmV7wdSdmSt3yjB4dqQaA58Td9"`
```

Comment:
Did you deploy this NFT and image? If not I would just clarify who did or that you are using it for educational purposes only

Reply:
yes I did it, I'll add its for educational purposes